### PR TITLE
pb-7035: fixed issue in the check to route the PXD/NFS to not nfs logic for restore.

### PR DESCRIPTION
**Pull request by siva-portworx**
**What type of PR is this?**
> Uncomment only one and also add the corresponding label in the PR:
>bug
**What this PR does / why we need it**:
```
pb-7035: fixed issue in the check to route the PXD/NFS to not nfs logic for restore.
```

**Does this PR change a user-facing CRD or CLI?**:
No

**Is a release note needed?**:
NA. 

**Does this change need to be cherry-picked to a release branch?**:
24.2.1 branch
**Testing:
Validate restore of following cases:**
1) restore from backup with CSI/PXD from NFS BL.
2) restore from backup with KDMP/PXD from NFS BL.
3) restoring from backup CSI/PXD from NFS BL with restoring only PXD and only CSI.

### CODE REVIEW

**Review by lalat-das — APPROVED** [review]
lgtm

File: pkg/applicationmanager/controllers/applicationrestore.go [hunk 1]
```diff
@@ -679,10 +675,10 @@ func (a *ApplicationRestoreController) restoreVolumes(restore *storkapi.Applicat
 		// Portworx driver restore is not supported via job as it can be a secured px volume
 		// to access the token, we need to run the driver.startRestore in the same context as the stork controller
 		// this check is equivalent to (if !nfs || (nfs && driverName == volume.PortworxDriverName))
-		if !nfs || hasPXDdriver {
+		for driverName, vInfos := range backupVolumeInfoMappings {
 			// Here backupVolumeInfoMappings is framed based on driver name mapping, hence startRestore()
 			// gets called once per driver
-			for driverName, vInfos := range backupVolumeInfoMappings {
+			if !nfs || (nfs && driverName == volume.PortworxDriverName) {
```
Comment (vikasit12) [inline pkg/applicationmanager/controllers/applicationrestore.go:681]: nit :: We can have a simplified check like
if !nfs || driverName == volume.PortworxDriverName {}

Comment (vsundarraj-px) [reply pkg/applicationmanager/controllers/applicationrestore.go:681]: Oops didnt realise the flag would regress like this.

Comment (siva-portworx) [reply pkg/applicationmanager/controllers/applicationrestore.go:681]: Yes, checking of driverName alone will working but I prefer check is gives clarity in understanding the logic :)

**Pull request merged**